internal/server/pre_reboot.go

```diff
@@ -0,0 +1,70 @@
+package server
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/coreos/airlock/internal/lock"
+)
+
+const (
+	// PreRebootEndpoint is the endpoint for requesting a semaphore lock.
+	PreRebootEndpoint = "/v1/pre-reboot"
+)
+
+// PreReboot is the handler for the `/v1/pre-reboot` endpoint.
+func (a *Airlock) PreReboot() http.Handler {
+	handler := func(w http.ResponseWriter, req *http.Request) {
+		logrus.Debug("got pre-reboot request")
+		if a == nil {
+			http.Error(w, errNilAirlockServer.Error(), 500)
+			return
+		}
+
+		nodeIdentity, err := validateIdentity(req)
+		if err != nil {
+			logrus.Errorln("failed to validate client identity: ", err)
+			http.Error(w, err.Error(), 400)
+			return
+		}
+		logrus.WithFields(logrus.Fields{
+			"group": nodeIdentity.Group,
+			"UUID":  nodeIdentity.UUID,
```

Comment:
minor: as logrus seems to be based on JSON style output should we just lower-case the key name?

Reply:
Ack, I'll rename it. I'm not sure I got the meaning of the first part of your comment though (the JSON part).

Reply:
JSON itself doesn't have a set standard but the most common ones I've seen used are camelCase & snake_case (fwiw [google's style guide chooses camelCase](https://google.github.io/styleguide/jsoncstyleguide.xml?showone=Property_Name_Format#Property_Name_Format))

Reply:
I see, thanks. Everywhere else here I've used lower_snake_case for identifiers, I'll try sticking to it.